soc/xtensa/intel_adsp/ace/power.c

```diff
@@ -383,10 +383,25 @@ void pm_state_exit_post_ops(enum pm_state state, uint8_t substate_id)
 
 #ifdef CONFIG_ARCH_CPU_IDLE_CUSTOM
 
+#include <pm_data.h>
+
 void arch_cpu_idle(void)
 {
+#if CONFIG_ADSP_DYNAMIC_CLOCK_SWITCHING
+	extern void adsp_clock_idle_entry(void);
+	extern void adsp_clock_idle_exit(void);
```

Comment:
An occasional extern declaration for a weakly-coupled internal function is sort of OK as a way to punch a hole into a separate subsystem.  But this is a real API with clear use and obvious applicability to the code here.  These declarations should be in a header under soc.h somewhere.